src/sidebar/components/test/menu-item-test.js

```diff
@@ -220,19 +233,43 @@ describe('MenuItem', () => {
 
     it('sets focus to the parent item when the submenu is closed via `closeMenu`', () => {
       const clock = sinon.useFakeTimers();
-      const wrapper = createMenuItem({
-        isSubmenuVisible: true,
-        submenu: <div>Submenu content</div>,
-      });
-      act(() => {
-        wrapper
-          .find('MenuKeyboardNavigation')
-          .props()
-          .closeMenu({ key: 'Enter' });
-      });
-      clock.tick(1);
-      assert.equal(document.activeElement.className, 'menu-item');
-      clock.restore();
+      try {
+        const wrapper = createMenuItem({
+          isSubmenuVisible: true,
+          submenu: <div role="menuitem">Submenu content</div>,
+        });
+        act(() => {
+          wrapper
+            .find('MenuKeyboardNavigation')
+            .props()
+            .closeMenu({ key: 'Enter' });
+        });
+        clock.tick(1);
+        assert.equal(document.activeElement.className, 'menu-item');
+      } finally {
+        clock.restore();
+      }
+    });
+
+    it('does not throw an error when unmounting the component before the focus timeout finishes', () => {
```

Comment:
I confirmed this test fails when commenting out the respective clearTimeout() call in the component